dlib/dnn/loss.h

```diff
@@ -3780,13 +3780,13 @@ namespace dlib
                             const double th = truth_box.rect.height() / (anchors[a].height + truth_box.rect.height());
 
                             // Scale regression error according to the truth size
-                            const double scale_box = 2 - truth_box.rect.area() / input_area;
+                            const double scale_box = options.lambda_box * (2 - truth_box.rect.area() / input_area);
 
                             // Compute the gradient for the box coordinates
-                            g[x_idx] = options.lambda_box * scale_box * (out_data[x_idx] - tx);
-                            g[y_idx] = options.lambda_box * scale_box * (out_data[y_idx] - ty);
-                            g[w_idx] = options.lambda_box * scale_box * (out_data[w_idx] - tw);
-                            g[h_idx] = options.lambda_box * scale_box * (out_data[h_idx] - th);
+                            g[x_idx] = scale_box * (out_data[x_idx] - tx);
+                            g[y_idx] = scale_box * (out_data[y_idx] - ty);
+                            g[w_idx] = scale_box * (out_data[w_idx] - tw);
+                            g[h_idx] = scale_box * (out_data[h_idx] - th);
```

Comment:
Just computing the scaling factor once... This might have no impact and be optimized away by the compiler though...